### PR TITLE
Allow rapid render in sandbox (#122828)

### PR DESCRIPTION
**Pull request by bpasero**
This PR is to bring rapid renderer into sandbox (and maybe in the future even support it in browsers but that is not the scope of this issue/PR).

@jrieken mainly for the changes in [`splash.ts`](https://github.com/microsoft/vscode/compare/ben/122828?expand=1#diff-f2b4f355aabe8b92d2c8a33a8f429e88d309dcebe5aa52de360e1abe1f12472b) which now is a core workbench thing (no longer contrib) and no longer writes to `rapid_render.json` directly but rather tells main to deal with it. The changes in `workbench.js` are smaller, mainly removed the sync fs read on startup because the splash data can now come in directly through the window config.

@aeschli for some cleanup and additions to theme service. I notice that we used to send a `vscode:changeColorTheme` IPC message from the parts splash to the main theme service to update something in `storage.json`. This is now cleaner because the splash is also going into `storage.json` and I felt that the theme main service could deal with all of it in one go. I did not want to introduce another service for this, even though splash data is arguable not fully theming related. I am open to suggestions if you want to change this though.

### CODE REVIEW

File: src/bootstrap-window.js [hunk 1]
```diff
@@ -57,6 +57,11 @@
 		const configuration = await preloadGlobals.context.resolveConfiguration();
 		performance.mark('code/didWaitForWindowConfig');
 
+		// Signal DOM modifications are now OK
```
Comment (bpasero) [inline src/bootstrap-window.js:60]: This is just moving up to draw as early as possible.

File: src/vs/code/electron-browser/workbench/workbench.js [hunk 2]
```diff
@@ -114,28 +115,15 @@
...
-			try {
-				data = JSON.parse(require.__$__nodeRequire('fs').readFileSync(configuration.partsSplashPath, 'utf8'));
-			} catch (e) {
-				// ignore
-			}
-		}
+		let data = configuration.partsSplash;
 
 		// high contrast mode has been turned on from the outside, e.g. OS -> ignore stored colors and layouts
-		const isHighContrast = configuration.colorScheme === 'hc' /* ColorScheme.HIGH_CONTRAST */ && configuration.autoDetectHighContrast;
```
Comment (bpasero) [inline src/vs/code/electron-browser/workbench/workbench.js:138]: I think this was actually a bug due to missing types.

File: src/vs/code/electron-sandbox/workbench/workbench.js [hunk 3]
```diff
@@ -112,5 +114,96 @@
 		return window.MonacoBootstrapWindow;
 	}
 
+	/**
+	 * @param {INativeWindowConfiguration & NativeParsedArgs} configuration
+	 */
+	function showPartsSplash(configuration) {
```
Comment (bpasero) [inline src/vs/code/electron-sandbox/workbench/workbench.js:120]: This is a 100% copy of the other one in `electron-browser`. Once we have enabled sandbox, the duplication will go away.

File: src/vs/platform/native/common/native.ts [hunk 4]
```diff
@@ -72,6 +72,8 @@ export interface ICommonNativeHostService {
 
 	setMinimumSize(width: number | undefined, height: number | undefined): Promise<void>;
 
+	saveWindowSplash(splash: IPartsSplash): Promise<void>;
```
Comment (bpasero) [inline src/vs/platform/native/common/native.ts:75]: This is maybe not the best service to put it, ideally we had a theme service that the window can talk to that travels to the main side. We can think about introducing a new service for this if we wanted to.

File: src/vs/workbench/electron-sandbox/splash.ts [hunk 5]
```diff
@@ -73,16 +67,22 @@ class PartsSplash {
 	}
 
 	private _savePartsSplash() {
-		const baseTheme = getThemeTypeSelector(this._themeService.getColorTheme().type);
+		this._savePartsSplashScheduler.trigger(() => this._doSavePartsSplash(), 800);
+	}
```
Comment (jrieken) [inline src/vs/workbench/electron-sandbox/splash.ts:71]: I don't believe that the `ThrottledDelayer` is an improvement. I see more code and more allocations

Comment (bpasero) [reply src/vs/workbench/electron-sandbox/splash.ts:71]: Removed it.

**Review by aeschli — APPROVED** [review]
LGTM! Thanks for adding all the types! Also the changes to the ThemeMainService are fine and IMO using the ICommonNativeHostService for saveWindowSplash is also ok.

**Pull request merged**